File: frontend/merp/src/types/accounting/AccountOutput.ts
export const ACCOUNT_TYPES = [
  "ASSET",
  "LIABILITY",
  "EQUITY",
  "REVENUE",
  "EXPENSE",
] as const;

export type AccountType = (typeof ACCOUNT_TYPES)[number];

export interface AccountOutput {
  id: number;
  parentAccountId: number | null;
  code: string;
  type: AccountType;
  name: string;
  description: string;
  receiveBalance: boolean;
  state: boolean;
}

File: backend/src/main/java/dev/mateorossello/merp/modules/accounting/dtos/AccountOutput.java
package dev.mateorossello.merp.modules.accounting.dtos;

import dev.mateorossello.merp.modules.accounting.models.AccountType;

public record AccountOutput(
    Long id,

    Long parentAccountId,

    String code,

    AccountType type,

    String name,

    String description,

    boolean receiveBalance,

    boolean state
) {}
